askametric/query_processor/query_processor.py

```diff
@@ -241,20 +251,41 @@ async def _get_final_answer_from_llm(self) -> None:
         self.cost += float(final_answer_llm_response["cost"])
         self.final_answer_prompt = prompt
 
+    @track_time(create_class_attr="timings")
+    async def _run_data_analysis(self) -> None:
+        try:
+            await self._get_best_tables_from_llm()
+            await self._get_best_columns_from_llm()
+            await self._get_sql_query_from_llm()
+            await self._get_final_answer_from_llm()
+
+        except Exception as e:
+            self.logger.error(f"Error processing query: {e}")
+            self.final_answer = (
+                "Sorry we faced an error while processing your query. "
+                "Please ask another question."
+            )
```

Comment:
Can we keep final_answer empty? User can know what happened by checking the status Enum, and they can choose what message to send to users if its internal error?

Reply:
Ok! Done!